serverless/src/samlCallback/handler.js

```diff
@@ -25,30 +26,24 @@ const getLaunchpadToken = (cookieString) => {
  */
 const samlCallback = async (event) => {
   const { body, headers } = event
-  const { Cookie } = headers
+  const { Cookie: headerCookie } = headers
   const { mmtHost } = getApplicationConfig()
 
   const params = new URLSearchParams(body)
   const path = params.get('RelayState')
 
-  const launchpadToken = getLaunchpadToken(Cookie)
-  const samlResponse = parseSaml(params.get('SAMLResponse'))
-  const { auid } = samlResponse
+  const launchpadToken = getLaunchpadToken(headerCookie)
 
-  // There appears to be a limitation in AWS to only allow sending 1 cookie, so we are sending
-  // 1 cookie with multiple values in a base 64 encoded json string.
-  let setCookie = `launchpadToken=${launchpadToken}; Secure; Path=/; Domain=.earthdatacloud.nasa.gov`
+  const edlProfile = await fetchEdlProfile(launchpadToken)
 
-  if (process.env.IS_OFFLINE) {
-    setCookie = `launchpadToken=${launchpadToken}; Path=/`
-  }
+  // Create JWT with launchpad token and edl profile
+  const jwt = createJwt(launchpadToken, edlProfile)
 
-  const location = `${mmtHost}/auth_callback?target=${encodeURIComponent(path)}&auid=${encodeURIComponent(auid)}`
+  const location = `${mmtHost}/auth-callback?target=${encodeURIComponent(path)}&jwt=${jwt}`
```

Comment:
Isn't this the same as passing the launchpad token in the URL, which is considered bad practice.   The JWT isn't really hiding it as it can be decoded by anyone.